### PR TITLE
Editorial: Explain `next` more accurately

### PR DESCRIPTION
**Pull request by gibson042**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,10 +70,6 @@ Return the code unit index of the most recently discovered break position, as an
 
 The `breakType` of the most recently discovered segment. If there is no current segment (e.g., a just-instantiated SegmentIterator, or one which has reached the end), or if the break type is "grapheme", then this will be `undefined`.
 
-For most programmers, the most important differences may be
-- Between `"none"` and everything else for word breaks (where `"none"` indicates that something is not a word)
-- Between `"soft"` and `"hard"` for line breaks (where `"soft"` indicates a line break opportunity, such as a space, and `"hard"` indicates a forced line break possibility, such as a `\n` character)
```
Comment (littledan) [inline README.md]: Not sure why this PR removes the description of breakType. Was this a mistake?

File: README.md [hunk 2]
```diff
@@ -70,10 +70,6 @@ Return the code unit index of the most recently discovered break position, as an
 
 The `breakType` of the most recently discovered segment. If there is no current segment (e.g., a just-instantiated SegmentIterator, or one which has reached the end), or if the break type is "grapheme", then this will be `undefined`.
 
-For most programmers, the most important differences may be
```
Comment (littledan) [inline README.md]: Removing this information from this point would probably make more sense if we merge the other PR. Also, I think the bulleted description is more clear than a big paragraph, so I don't understand why that change was made.

Comment (gibson042) [reply README.md]: I was trying to move the details of `breakType` out of this specific getter. The latest update dedicates a specific section to the iteration result fields.

**Comment by littledan** [comment]
The current explainer and specification work to define `next` accurately. Is this PR still needed?

**Comment by gibson042** [comment]
The current definition is much improved, but still includes a TODO. I've cleaned it up this PR to resolve that.

**Pull request merged**